### ISSUE TITLE
Random issues

### ISSUE THREAD

**Issue by Morganamilo** [pi-kvm/yay-kit#1]
Use `systemd-sysusers(8)` instead of an install script.

Root is defined to have a UID of 0. The root account can technically have any name, so your check is incorrect.

Why even have the root check anyway? You're calling sudo so as long as my user is in wheel I don't need to be root.

`pacman --noconfirm -Syy` are you asking for partial upgrades? Seriously why is this here?

`yay --noconfirm` are you asking to be hacked?

You do a lot of weird specific stuff in `pkg-install` stuff that is unsafe and other stuff that's just specific to you. I can understand the usefulness of `yay-user` (minus the root check)

**Comment by mdevaev** [comment]
> Use systemd-sysusers(8) instead of an install script.
OMG another new systemd crutch :)

> Root is defined to have a UID of 0. The root account can technically have any name, so your check is incorrect.
In fact no one user will break the operation system because `root` is a factical standard and many programs count on it.

> Why even have the root check anyway? You're calling sudo so as long as my user is in wheel I don't need to be root.
Maybe I should add a more strict check.

> Seriously why is this here?
> are you asking to be hacked?
> weird specific stuff
You're found this repository but you don't know what is it for. These scripts are needed for automatic build the operating system inside docker for raspberry pi (not container, real OS). This is a not common package for `yay` (AUR contains other one). Also this is an experimental project so when we run it in production we'll create own binary repository for Arch Linux ARM with our packages.

**Comment by Morganamilo** [comment]
>You're found this repository but you don't know what is it for

I found this package on the AUR and assumed it was a wrapper to let you run yay as root. Which as a general thing I can maybe find useful.

And as the rules AUR's of submission  state:

>Make sure the package you want to upload is useful. Will anyone else want to use this package? Is it extremely specialized?

Currently, as you say this package is "experimental" and for container stuff (I'd argue the idea itself is flowed and the `-Syy` can still screw you over). As it stands any normal user trying to use this opening themselves up to partial upgrades and malicious pkgbuilds. If you do not intend for people to use this package then I suggest you remove it from the AUR.

**Comment by mdevaev** [comment]
> wrapper to let you run yay as root

And it's working. You can use it without any problems. Okay, maybe I can check for ther `wheel` group.

> As it stands any normal user trying to use this opening themselves up to partial upgrades and malicious pkgbuilds

Doesn't it seem strange to you to run a script whose purpose you don't know? You last note only applies to `pkg-install` and the only purpose of this script is to install a package unconditionally without asking questions. If you care about safety just don't use it. The maximum that I can do is to add a warning before using this script, but I will not remove the package from the AUR.

**Comment by eli-schwartz** [comment]
> The maximum that I can do is to add a warning before using this script, but I will not remove the package from the AUR.

I can and will.

**Comment by mdevaev** [comment]
This package made it easier to work with AUR, with which it is impossible to work from the root.

@eli-schwartz I have no words. You guys are awesome. Friendly community.

**Comment by eli-schwartz** [comment]
A shellscript that does nothing but execute `sudo -u anotheruser yay` does not make it "easier to work with the AUR, since 63K registered users and no one knows how many tens of thousands of unregistered ones use the AUR just fine as their login user. Arch has never recommended running without any configured users. There is even a "nobody" user configured by default which can be used.

Either way that isn't actually the point. The point is that a one-liner script to run a command with sudo is a violation of the rules of submission as you were already told.

The fact that it was solely designed to do dangerous security things was simply icing on the cake.

**Comment by mdevaev** [comment]
> The point is that a one-liner script to run a command with sudo is a violation of the rules of submission as you were already told.

No, it doesn't. This package solved the practical task: created a user and made it possible to use yay from root. There are many cases of why this is necessary and why a dedicated system user may be required to build a package. For embedded systems for example.

In addition, you deleted my package after I've added warning message to `pkg-install`. You don't think that you are doing wrong, without understanding the situation?

**Comment by eli-schwartz** [comment]
Embedded systems have no more need for this package than any other, and you know it. You can deploy site-specific user configuration in a variety of ways... or consider that `sudo -u nobody env HOME=/tmp/yayhome yay -Syu` should work fine too.

So... first you claim the purpose of the package is only to let *you* build an rpi image in docker (thus not "useful to more than one person"), then you claim that it's necessary in "many cases [...] for embedded systems for example" (which is still the rpi image, and not a proof of "many cases"). Okay -- I believe you that it's only useful for unsupported architectures.

If you cannot find any justification for the package other than "for embedded systems", then you should also consider that the AUR is only intended for packages that can be installed and do something useful on the officially supported x86_64 platform -- other arch support is considered to be a bonus and available at the maintainer's discretion. So this is already becoming a case of special circumstances.

A package that is:
- not useful for x86_64 archlinux
- not useful to more than one person
- heavily violates all system stability recommendations (with or without warning the user, but in this case it is "without")
- performs a one-line task, practical or otherwise

has too many things going against it.

> In addition, you deleted my package after I've added warning message to `pkg-install`. You don't think that you are doing wrong, without understanding the situation?

The warning message does not warn the user that this script is unnecessary and only useful to the original uploader.

The warning message does not warn the user that use of the script automatically breaks their system no matter what (because it runs pacman -Syy and then installs a package without doing a full system update).

The warning message does not warn the user that use of the script deletes a lot of cache data from /var/cache/pacman/pkg even though most users do not want to delete and then redownload the same packages many times, for example because makedepends get uninstalled between builds (or the builds happen in a clean chroot with a persistent cache).

**Comment by mdevaev** [comment]
> Embedded systems have no more need for this package than any other, and you know it

Please do not tell me what I know and what I do not.

> It's only useful for unsupported architectures.
> ... not useful for x86_64 archlinux

No. If I want to build any package inside a container on any architecture, I will not be able to do this from a privileged user.

> heavily violates all system stability recommendations (with or without warning the user, but in this case it is "without")

Wrong. A warning has been added after request. Let's not practice windows approach. The user is able to decide what to do with his operating system.

>  this script is unnecessary and only useful to the original uploader

This is only your opinion, because maybe you didn't encounter tasks that require such a solution.

If you carefully re-read the entire dialogue from the very beginning - I was not against making changes to the package to meet any requirements. But instead of a constructive discussion, your friend called you. And you, instead of figuring out what this package is for and offering a good solution, just prefer to remove it. You interpret the rules the way you want, so as not to upset your friend. Your behavior is toxic and unprofessional.
